src/java/org/apache/cassandra/service/reads/AbstractReadExecutor.java

```diff
@@ -247,60 +258,47 @@ public void maybeTryAdditionalReplicas()
 
     static class SpeculatingReadExecutor extends AbstractReadExecutor
     {
-        private volatile ReplicaLayout.ForToken replicaLayout;
         private volatile boolean speculated = false;
 
         public SpeculatingReadExecutor(ColumnFamilyStore cfs,
                                        ReadCommand command,
                                        ReplicaLayout.ForToken replicaLayout,
                                        long queryStartNanoTime)
         {
-            super(cfs, command, replicaLayout, queryStartNanoTime);
-            this.replicaLayout = replicaLayout;
-        }
-
-        public ReplicaLayout.ForToken replicaLayout()
-        {
-            return replicaLayout;
-        }
-
-        public void executeAsync()
-        {
-            EndpointsForToken initialReplicas = replicaLayout().selected();
-
-            if (handler.blockfor < initialReplicas.size())
-            {
-                // We're hitting additional targets for read repair.  Since our "extra" replica is the least-
```

Comment:
This comment was lost after your changes and I needed it to answer the question of why some of the time we make two data requests